workflow-manager/main.go

```diff
@@ -177,29 +316,27 @@ func main() {
 		}
 	}
 
-	interval := aggregationInterval(aggregationPeriodParsed, gracePeriodParsed)
+	interval := aggregationInterval(config.clock, config.aggregationPeriod, config.gracePeriod)
 	log.Printf("looking for batches to aggregate in interval %s", interval)
 	aggregationBatches = withinInterval(aggregationBatches, interval)
 	aggregationMap := groupByAggregationID(aggregationBatches)
-	r := retry.Retry{
-		Identifier: "Aggregation",
-		Retryable: func() error {
-			return launchAggregationJobs(context.Background(), aggregationMap, interval)
-		},
-		ShouldRequeue: wferror.IsTransientErr,
-		// 5 is an arbitrary number... it seemed right
-		MaxTries: 5,
-		// No need to wait between retries
-		TimeBetweenTries: 0,
-	}
-
-	if err := r.Start(); err != nil {
+	err = enqueueAggregationTasks(
+		context.Background(),
```

Comment:
You can get the Background context anywhere, no need to pass per fn param. Also keep the context as narrow as possible and required